src/pages/MovieDetails.vue

```diff
@@ -15,44 +36,211 @@ export default defineComponent({
   },
   data() {
     return {
-      movieDetails: null,
-      isLoading: true,
-      errorMessage: null,
+      movie: null,
+      movieIsLoading: true,
+      movieError: null,
+      date: new Date(),
     };
   },
   computed: {
+    ...mapState(useSeancesStore, [
+      "seances",
+      "seancesIsLoading",
+      "seancesError",
+      "getSeancesErrorMessage",
+    ]),
+    getMovieErrorMessage() {
+      return (
+        this.movieError?.message ||
+        "We are sorry, but the movie cannot be displayed"
+      );
+    },
     getMovieTitle() {
-      return this.movieDetails.title;
+      return this.movie.title;
+    },
+    getMovieReleaseYear() {
+      return this.movie.release_date.substring(0, 4);
+    },
+    getImagePath() {
+      return this.movie.poster_url;
+    },
+    getStyledImage() {
+      return { background: "url(" + this.getImagePath + ") center / cover" };
+    },
+    getFormattedWeekdayAndDate() {
+      const weekday = this.date.toLocaleDateString("en-US", {
+        weekday: "long",
+      });
+      const formattedDate = this.date.toLocaleDateString("en-GB");
+      return `${weekday} ${formattedDate}`;
+    },
+    formattedDate() {
+      return this.date.toISOString().substring(0, 10);
+    },
+  },
+  watch: {
+    date(newDate, oldDate) {
+      if (newDate !== oldDate) {
+        this.getSeances(this.formattedDate);
+      }
     },
   },
   methods: {
-    async getMovieDetails() {
-      this.isLoading = true;
+    async getMovie() {
+      this.movieIsLoading = true;
       try {
         const respData = await getMovieById(this.movieId);
-        this.movieDetails = respData.data;
+        this.movie = respData.data;
       } catch (error) {
-        this.errorMessage = error.message;
+        this.movieError = error;
```

Comment:
another option to consider, you could define your error message here and pass it to the store at this point then you could load it up in your error component directly from the store 💡

Reply:
Basically I decided to keep the movie state in the local state of the MovieDetails component, cuz it is the only component where there is an api call to get the single movie data.
I get seances data from the pinia store (there are api calls to the seances form multiple components) and in this case I get error message from the store.